### ISSUE TITLE
Exceptions in Blade templates

### ISSUE THREAD

**Issue by swayok** [klesun/deep-assoc-completion#147]
Hi. I'm getting many exceptions with same trace (only tree.length and psi.length=17 differ):
```
java.lang.AssertionError: File text mismatch: tree.length=16; psi.length=17; this=PHP file; vp=Multi root injected file 'new_user_request_form.blade.php' 
	at com.intellij.psi.impl.source.PsiFileImpl.getText(PsiFileImpl.java:288)
	at org.klesun.deep_assoc_completion.completion_providers.ArgCstPvdr.addCompletions(ArgCstPvdr.java:66)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:26)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:150)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:141)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:77)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:26)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:150)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:119)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.performCompletion(CompletionServiceImpl.java:55)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:827)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:812)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$null$5(CodeCompletionHandlerBase.java:326)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:997)
	at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:170)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$6(CodeCompletionHandlerBase.java:318)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$null$0(CompletionThreading.java:95)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:91)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by klesun** [comment]
Hi, thanks for the report. Could you, please, post a minimal blade file that reproduces the issue?

**Comment by swayok** [comment]
Here it is. I'm using many custom blade directives added to `Settings -> Languages & Frameworks -> PHP -> Blade -> Directives`. Maybe it is related.

```
<script type="text/html" id="user-request-form-tpl">
    <div class="admin-form popup-basic popup-lg" id="user-requests-form-container">
        <form class="form-horizontal" id="user-requests-form" enctype="multipart/form-data"
              action="{{ route('m_user_request_create') }}" method="POST">
            {!! csrf_field() !!}
            <div class="panel">
                <div class="panel-heading pbn pt10 text-center">
                    <span class="panel-title">@trans('management_area.user_requests.form.header_for_creation')</span>
                </div>

                <div class="panel-body">
                    <input type="hidden" name="building_id" value="@jsEcho(it.building_id)">
                    <div class="form-group">
                        <label class="control-label col-sm-3" for="select-type">
                            @trans('management_area.user_requests.form.input.type')
                        </label>
                        <div class="col-sm-9">
                            <div class="field">
                                <select name="user_requests_type_id" class="form-control selectpicker" id="select-type">
                                    @jsForEach(it.types as typeInfo)
                                        <option
                                            value="@jsEcho(typeInfo.id)"
                                            data-sets-execute-at-automatically="@jsEcho(typeInfo.set_execution_date_automatically ? '1' : '0')"
                                            data-executor-required-on-accept="@jsEcho(typeInfo.is_executor_required_on_accept ? '1' : '0')"
                                            data-is-apartment-required-on-create="@jsEcho(typeInfo.is_apartment_required_on_create ? '1' : '0')"
                                            data-require-file="@jsEcho(typeInfo.requires_attached_file ? '1' : '0')"
                                        >
                                            @jsEcho(typeInfo.title)
                                        </option>
                                    @jsEndForEach
                                </select>
                            </div>
                            <div id="user-request-type-info-container" class="mt10 posr pl20">
                                <div style="position: absolute; left: 0;">
                                    <i class="glyphicon glyphicon-info-sign text-info"></i>
                                </div>
                                <div>
                                    @jsForEach(it.types as typeInfo)
                                        <div class="user-request-type-info hidden text-muted" data-request-type="@jsEcho(typeInfo.id)">
                                            @jsEcho(typeInfo.description)
                                            <p>
                                                @jsIf(typeInfo.set_execution_date_automatically)
                                                    @html('management_area.user_requests.form.input.type_execution_date_automatic', [
                                                        'days' => '{{= typeInfo.days_for_execution }}'
                                                    ])
                                                @jsElse
                                                    @html('management_area.user_requests.form.input.type_execution_date_manual', [
                                                        'days' => '{{= typeInfo.days_for_execution }}'
                                                    ])
                                                @jsEndIf
                                            </p>
                                            @jsIf(typeInfo.files && typeInfo.files.length)
                                                <div class="user-request-type-files">
                                                    <i class="glyphicon glyphicon-paperclip mr5"></i>
                                                    @jsForEach(typeInfo.files as index => fileInfo)
                                                        <a
                                                            href="@jsEcho(fileInfo.url)"
                                                            target="_blank"
                                                            class="user-request-type-file"
                                                        >
                                                            @jsEcho(fileInfo.name)
                                                        </a>@jsEcho(index !== fileInfo.length - 1 ? ',' : '')
                                                    @jsEndForEach
                                                </div>
                                            @jsEndIf
                                        </div>
                                    @jsEndForEach
                                </div>
                            </div>
                        </div>
                    </div>

                    <div class="form-group">
                        <label class="control-label col-sm-3" for="input-subject">
                            @trans('management_area.user_requests.form.input.subject') *
                        </label>
                        <div class="col-sm-9">
                            <div class="field">
                                <input type="text" id="input-subject" name="subject" class="form-control">
                            </div>
                        </div>
                    </div>

                    <div class="form-group">
                        <label class="control-label col-sm-3" for="input-description">
                            @trans('management_area.user_requests.form.input.description')
                        </label>
                        <div class="col-sm-9">
                            <div class="field">
                                <textarea id="input-description" name="description" class="form-control" rows="4"></textarea>
                            </div>
                        </div>
                    </div>

                    <div class="form-group">
                        <label class="control-label col-sm-3" for="input-comment">
                            @trans('management_area.user_requests.form.input.comment')
                        </label>
                        <div class="col-sm-9">
                            <div class="field">
                                <textarea id="input-comment" name="comment" class="form-control" rows="3"></textarea>
                            </div>
                        </div>
                    </div>

                    <div class="section-divider mbn mt20">
                        <span>@trans('management_area.user_requests.form.input.user_info_section')</span>
                    </div>

                    <div class="form-group mt20">
                        <label class="control-label col-sm-3" for="select-apartment">
                            @trans('management_area.user_requests.form.input.apartment')
                        </label>
                        <div class="col-sm-9">
                            <div class="field">
                                <select name="apartment_owner_id" class="form-control selectpicker" id="select-apartment"
                                data-live-search="true">
                                    <option value="">&mdash;</option>
                                    @jsForEach(it.apartments_owners as owner)
                                        <option value="@jsEcho(owner.id)">
                                            @jsEcho(owner.apartment_type)
                                            @jsEcho(owner.number):
                                            @jsEcho(owner.surname) @jsEcho(owner.name) @jsEcho(owner.mid_name)
                                        </option>
                                    @jsEndForEach
                                </select>
                            </div>
                        </div>
                    </div>

                    <div class="form-group">
                        <label class="control-label col-sm-3" for="input-contact-phone">
                            @trans('management_area.user_requests.form.input.contact_phone')
                        </label>
                        <div class="col-sm-9">
                            <div class="field">
                                <input
                                    type="text"
                                    id="input-contact-phone"
                                    name="contact_phone"
                                    class="form-control"
                                    data-mask="+7 (999) 999-99-99"
                                >
                            </div>
                        </div>
                    </div>

                    <div class="section-divider mtn mb20">
                        <span>@trans('management_area.user_requests.form.input.execution_info_section')</span>
                    </div>

                    @can(\App\Management\ManagementAccessManager::USER_REQUESTS . '.set_executor')
                        <div class="form-group">
                            <label class="control-label col-sm-3" for="user-request-form-select-executor">
                                @trans('management_area.user_requests.form.input.executor')
                            </label>
                            <div class="col-sm-9">
                                <div class="field">
                                    <select
                                        name="executor_id"
                                        class="form-control selectpicker"
                                        id="user-request-form-select-executor"
                                        data-live-search="true"
                                    >
                                        <option value="">&mdash;</option>
                                    </select>
                                </div>
                            </div>
                        </div>
                    @endcan

                    @can(\App\Management\ManagementAccessManager::USER_REQUESTS . '.set_execute_at')
                        <div class="form-group" id="user-request-form-input-execute-at-container">
                            <label class="control-label col-sm-3" for="user-request-form-input-execute-at">
                                @trans('management_area.user_requests.form.input.execute_at')
                            </label>
                            <div class="col-sm-9">
                                <div class="field">
                                    <div class="input-group date w200">
                                        <input
                                            type="text"
                                            class="form-control w200"
                                            name="execute_at"
                                            id="user-request-form-input-execute-at"
                                            data-min-date="{{ date('Y-m-d') }}"
                                        >
                                        <span class="input-group-addon cursor"><i class="fa fa-calendar"></i></span>
                                    </div>
                                </div>
                            </div>
                        </div>
                    @endcan

                    <div class="section-divider mb30">
                        <span>@trans('management_area.user_requests.form.input.images')</span>
                    </div>

                    <div id="user-request-file-attachment-required" class="text-center text-muted fw600 mb30 collapse">
                        <span>@html('frontend.user_requests.form.errors.file.required')</span>
                    </div>

                    <div class="mb15">
                        <p class="text-danger">@html('frontend.user_requests.form.file_size_warning')</p>
                        <p>@html('frontend.user_requests.form.image_upload_fail_warning')</p>
                    </div>

                    <div class="form-group">
                        @for($i = 1; $i <= \App\Db\UserRequest\UserRequest::IMAGES_COUNT; $i++)
                            <div class="col-sm-4">
                                @include('management.ui.image_upload_dotjs_tpl', ['inputName' => 'image' . $i])
                            </div>
                        @endfor
                    </div>

                    <div class="section-divider mb30">
                        <span>@trans('management_area.user_requests.form.input.files')</span>
                    </div>

                    <div class="mb15">
                        <p class="text-danger">@html('frontend.user_requests.form.file_size_warning')</p>
                        <p>@html('frontend.user_requests.form.file_type_warning_documents')</p>
                    </div>

                    <div class="form-group">
                        @for($i = 1; $i <= \App\Db\UserRequest\UserRequest::FILES_COUNT; $i++)
                            <div class="col-sm-12">
                                @include('management.ui.file_upload_dotjs_tpl', ['inputName' => 'file' . $i])
                            </div>
                        @endfor
                    </div>

                </div>

                <div class="panel-footer text-right">
                    <button type="button" class="button dark pull-left modal-close" onclick="return false">
                        @trans('management_area.cancel')
                    </button>
                    <button type="submit" class="button btn-success">
                        @trans('management_area.user_requests.form.create')
                    </button>
                </div>
            </div>
        </form>
    </div>
</script>

<script type="text/html" id="user-request-form-executors-options-tpl">
    @jsForEach(it.executors as executor)
    <option
        value="@jsEcho(executor.id)"
        @jsIf(!executor.is_active)
            data-content="<span class='text-danger' title='@trans('management_area.user_requests.filter.executor_disabled')'>@jsEcho(executor.label)</span>"
        @jsEndIf
    >
        @jsEcho(executor.label)
    </option>
    @jsEndForEach
</script>

<script type="application/javascript">
    $(function () {
        UserRequests.creationFormValidationSettings = {
            rules: {
                type: {
                    required: true
                },
                subject: {
                    required: true,
                    minlength: 5
                }
            },
            messages: {
                type: {
                    required: '{!! addslashes(trans('management_area.user_requests.form.errors.type.required')) !!}'
                },
                subject: {
                    required: '{!! addslashes(trans('management_area.user_requests.form.errors.subject.required')) !!}',
                    minlength: '{!! addslashes(trans('management_area.user_requests.form.errors.subject.min', ['min' => 5])) !!}'
                },
                file1: {
                    required: '{!! addslashes(trans('frontend.user_requests.form.errors.file.required')) !!}'
                },
                apartment_owner_id: {
                    required: '{!! addslashes(trans('management_area.user_requests.form.errors.apartment_owner_id.required_if')) !!}'
                },
                executor_id: {
                    required: '{!! addslashes(trans('management_area.user_requests.form.errors.executor_id.required_if')) !!}'
                }
            }
        };
    });
</script>
```

**Comment by klesun** [comment]
Thanks a lot. On it...

**Comment by klesun** [comment]
I hope this release will fix your error - [2020.02.17.001](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/update/78420)

Did not manage to reproduce the issue without the directives - just added a try catch for the built-in function error. It appears to me as if issue is in the blade plugin itself.
